src/query/frontend/ast/ast.hpp

```diff
@@ -2093,6 +2114,9 @@ class CypherQuery : public memgraph::query::Query, public utils::Visitable<Hiera
   memgraph::query::SingleQuery *single_query_{nullptr};
   /// Contains remaining queries that should form and union with `single_query_`.
   std::vector<memgraph::query::CypherUnion *> cypher_unions_;
+  /// Index hint
+  std::vector<memgraph::query::IndexHint> index_hints_;
```

Comment:
put TODO message that if any other pre-query directives are to be added, then a super class holder -> something like PreprocessDirectives - needs to be added

Reply:
because we might want to add more except the indexing hints

Reply:
Added